Integrations/ThreatQ/ThreatQ.py

```diff
@@ -0,0 +1,899 @@
+
+    res = tq_post_request('/attachments', params)
+    if res.status_code >= 400:
+        return_error('Could not upload file.')'''
+
+
+def update_status_command(ioc_id, status):
+    if isinstance(ioc_id, str) and not ioc_id.isdigit():
+        return_error("Invalid argument for indicator ID.")
+
+    search_by_id_command("indicator", ioc_id, returns_output=False)  # returns error if object was not found
+
+    url_suffix = "/indicators/{0}".format(ioc_id)
+    params = {"status": status}
+
+    tq_request("PUT", url_suffix, params)
+    demisto.results("Successfully updated indicator #{0}'s status to {1}.".format(ioc_id, status))
+
+
+def get_ip_reputation(ip):
+    validate_ioc(ip, 'ipv4')  # todo: add ipv6 when there is a validation
+    generic = {"Address": ip}
+    raw, dbot = get_ioc_reputation(ip, 'ip')
+    entry_context = set_ioc_entry_context('ip', raw, dbot, generic)
+    readable = build_readable("Search results for IP {0}".format(ip), "indicator", raw)
+    return_outputs(readable, entry_context, raw)
+
+
+def get_url_reputation(url):
+    validate_ioc(url, 'url')
+    generic = {"Data": url}
+    raw, dbot = get_ioc_reputation(url, 'ip')
+    entry_context = set_ioc_entry_context('url', raw, dbot, generic)
+    readable = build_readable("Search results for URL {0}".format(url), "indicator", raw)
+    return_outputs(readable, entry_context, raw)
+
+
+def get_email_reputation(email):
+    validate_ioc(email, 'email')
+    generic = {"Address": email}
+    raw, dbot = get_ioc_reputation(email, 'email')
+    entry_context = set_ioc_entry_context('email', raw, dbot, generic)
+    readable = build_readable("Search results for email address {0}".format(email), "indicator", raw)
+    return_outputs(readable, entry_context, raw)
+
+
+def get_domain_reputation(domain):
+    # fmt = validate_ioc(domain, 'domain')  # todo: add validation
+    generic = {"Name": domain}
+    raw, dbot = get_ioc_reputation(domain, 'domain')
+    entry_context = set_ioc_entry_context('domain', raw, dbot, generic)
+    readable = build_readable("Search results for domain {0}".format(domain), "indicator", raw)
+    return_outputs(readable, entry_context, raw)
+
+
+def get_file_reputation(file):
+    fmt = validate_ioc(file, 'md5', 'sha1', 'sha256')
+    generic = createContext({
+        "MD5": file if fmt == 'md5' else None,
+        "SHA1": file if fmt == 'sha1' else None,
+        "SHA256": file if fmt == 'sha256' else None
+    }, removeNull=True)
+    raw, dbot = get_ioc_reputation(file, 'file')
+    entry_context = set_ioc_entry_context('file', raw, dbot, generic)
+    readable = build_readable("Search results for file {0}".format(file), "indicator", raw)
+    return_outputs(readable, entry_context, raw)
+
+
+''' EXECUTION CODE '''
+handle_proxy()
+command = demisto.command()
+LOG('command is {0}'.format(demisto.command()))
+try:
+    args = demisto.args()
+    if command == 'test-module':
+        test_module()
+    elif command == 'tq-search-by-name':
```

Comment:
the commands should start with "threatq".
This also applies to all the other commands